sidecar_ecs/sidecar_aws_resources.tf

```
@@ -0,0 +1,95 @@
+locals {
+  keyData = base64decode(var.container_registry_key)
+
+  repoCreds = jsonencode({
+    username = "_json_key"
+    password = "${local.keyData}"
+  })
+  sidecarSecret = jsonencode({
```

I tend to think that calling it `sidecarCredentials` is a better fit as well.
```suggestion
  sidecarCredentials = jsonencode({
```

---

Just realized this variable is not used throughout the rest of the code as you stored the credentials in SSM.